src/com/sap/piper/tools/neo/NeoCommandHelper.groovy

```diff
@@ -8,19 +8,22 @@ class NeoCommandHelper {
     private Script step
     private DeployMode deployMode
     private Map deploymentConfiguration
+    private Set extensions
     private String user
     private String password
     private String source
 
     //Warning: Commands generated with this class can contain passwords and should only be used within the step withCredentials
     NeoCommandHelper(Script step, DeployMode deployMode, Map deploymentConfiguration,
+                    Set extensions,
                     String user, String password, String source) {
         this.step = step
         this.deployMode = deployMode
         this.deploymentConfiguration = deploymentConfiguration
         this.user = user
         this.password = password
         this.source = source
+        this.extensions = extensions ?: (Set)[]
```

Comment:
> I think that the way to declare an empty Set is [].toSet()

Why do you think that is "the standard"? What is bad with the cast/coerce approach. I mean having a certain method does not mean that this is "the standard". By the way: there are even more ways: `x as Set` would eg. also be possible. All roads lead to Rome.

Reply:
Yes, the Groovy cast would be using ` as`, I would do the casting like this in general. The way as it was done is Java.
In this case, there is 3 ways to do it in Groovy:
```
Set extensions = []
def extensions = [].toSet()
def extensions = [] as Set
``
All of them would be ok, and they are the standard way to do it in Groovy.

Reply:
@alejandraferreirovidal

First of all we agree that there is no explicit cast necessary in case the variable is typed.

> In this case, there is 3 ways to do it in Groovy:

I know more than 3 ways ...

In all cases (mentioned by you above and the good old cast) there is an attempt to coerce `foo` to a set. I do not get what is bad with the cast approach. On the other hand I do - of course - not say the other approaches are inferior. Who is making the standards? Can we read somewhere which approaches are considered to be standard and which one are not?

The discussion here is somehow weird.